Adapt the UI to smaller screen sizes

stasm opened the pull request:
In this PR I tried to make the UI work better on smaller screens.

Using Bootstrap's fluid container helps make the most of the available screen estate.  I reduced the total number of columns in the UI by moving the non-supply cards under the common ones.  I also use `col-xs-*` grid classes to keep the current layout on smaller screens (phones and tablets).  The previously used `col-md-*` classes would cause the grid cells to stack vertically below 992px of width, i.e. on all phones and tablets.

Here's what the game looks like now on handhelds:

**iPad (1024x768):**

![dominion-game-yjtannsry7uppdt6s ipad](https://cloud.githubusercontent.com/assets/265818/17278285/e04e766c-5759-11e6-8a70-9c5ca2b8ef17.png)

**iPhone 5 (568x320):**

![dominion-game-yjtannsry7uppdt6s iphone 5](https://cloud.githubusercontent.com/assets/265818/17278286/e04ed526-5759-11e6-9882-99faed6d7266.png)

stasm committed:
Move non-supply cards under common cards

Use Bootstrap's fluid container to make the most of the available screen
estate.  Reduce the total number of columns in the UI by moving the non-supply
cards under the common ones.  Use `col-xs-*` grid classes to keep the layout on
smaller screens (phones and tablets).  The previously used `col-md-*` classes
would cause the grid cells to stack vertically below 992px of width.

stasm committed:
Use text-overflow: ellipsis for long card names

stasm committed:
Move Destroy Game button to the Info modal

jcbantuelle commented:
The UI is certainly lacking and needs somebody with a better handle on front-end capabilities than I have, particularly in mobile. I really appreciate this work and I love almost all of the styling changes in here. My one concern here is how the non-supply area can pile up with a ton of cards in certain situations. I worry that moving that column makes the experience more problematic on desktops like in these screenshots:

![Desktop Non-Supply 1](https://cloud.githubusercontent.com/assets/1465895/17302330/5ca99dc2-57e1-11e6-852b-bbf477c751c6.png)

![Desktop Non-Supply 2](https://cloud.githubusercontent.com/assets/1465895/17302324/4dfd2cf8-57e1-11e6-8975-44872e8e0555.png)

I wouldn't want to disrupt the ability for somebody on a desktop to be able to see all of the cards without needing to scroll.

Thank you so much for your effort here, I'm excited to have UI improvements. Let me know your thoughts on this, and if you have any other ideas for how to accommodate multiple devices.

stasm commented:
Thanks for the feedback, @jcbantuelle.  I see your point about the non-supply cards.  I'll try to come up with a solution!

stasm committed:
Move non-supply cards under commons only on smaller screens

Three-column layout for all cards is better for larger screens.

stasm commented:
@jcbantuelle -- I changed how the layout reacts to screen sizes.  The non-supply column now wraps under the commons on phones, tablets and small desktops.  On large desktops the three-column layout is preserved.

This was much easier to implement once I changed the order of columns.  The new order is: kingdom, commons, non-supply.  I hope that's fine.

When there are no non-supply cards the UI has a gap on larger screens:

![localhost-3000-game-xxpqoc8pyojth5tlh](https://cloud.githubusercontent.com/assets/265818/17357660/33e1d214-595e-11e6-8980-f25e4c6a28af.png)

Does it look okay to you?

stasm commented:
Nevermind, I figured out how to keep the old layout (non-supply cards on far left) with Bootstrap's `col-*-push-*` and `col-*-pull-*` classes (http://getbootstrap.com/css/#grid-column-ordering).  I'll rebase this branch on top of the current master to support landmarks, too.

stasm committed:
Reorder card columns on large screens

Use Bootstrap's col-*-push-* and col-*-pull-* classes to reorder the non-supply
cards column on large screens and move it to the far-left side.  See
http://getbootstrap.com/css/#grid-column-ordering.

jcbantuelle commented:
I absolutely love these updates. I'm very excited to merge them in, but just have one concern right now that hopefully you'll have an idea on how to account for. Between Adventures and Empries, there are a lot of stacks that can accumulate tokens of various types, and Empires in particular has Split Piles that cause some very lengthy names as well.

I've attached a screenshot of what I'm seeing, it seems like when the names and tokens run too long there's not enough space for the kingdom card column. This kind of styling is definitely not my strong suit, but if we can somehow bump the amount of space given to the `kingdom-cards` column, I think we should be all set.

Let me know your thoughts, and thank you so very much for your contributions. I think this is going to be a much better styling and markup than what I had thrown together before.

![Long Kingdom Card Names](https://cloud.githubusercontent.com/assets/1465895/17422826/234e5d00-5a7b-11e6-8cbb-967fd70545bc.png)

stasm committed:
Make Kingdom cards column wider

stasm committed:
Disallow text wrapping in stack names

stasm commented:
@jcbantuelle great feedback, thanks!  I made the Kingdom cards column wider relative to the other two.  The proportions are now: 4-5-3.  This leaves little space for the common cards but admittedly they're almost always the same from game to game and the icons make them easy to tell apart anyways.  I'm a bit worried that in rare cases (just before the layout switches to the small/medium variant) some tokens on the common cards might not be visible.  This might happen when both Aqueduct and Tax are in play.

I also forced stack names to never line-break: if the name is too long for the width of the column the excess is clipped and an ellipsis is shown instead. With the enabled, it might actually make sense to go back to 4-4-4 columns on large screens. What do you think?

I'm still learning all the new cards in Adventures and Empires.  I tried to test as much as I can using the trick you showed me before. Let me know what you think.

jcbantuelle commented:
This is a huge improvement to the app, thank you!

jcbantuelle commented:
On an unrelated note, I got frustrated enough with Familiar, Black Market, and Possession that I commented them out of `card_list.js`. If you still want to play with them, you'll need to remove those commented out lines.

stasm commented:
Haha, yeah, I've seen the commit :)  I understand the frustration. I learned to tolerate these cards as part of the game.  It's a tough love though.  Maybe there should be an additional checkbox in the lobby for including/excluding just these 3 cards?

BTW I have a couple of further improvements on my mind (e.g. notifications when it's your turn) that I'd love to get your input on.  What's the best way to discuss them? Should I file separate issues? Maybe you'd consider setting up a gitter.im chat for this repo?

jcbantuelle commented:
I haven't used gitter.im before, but I just started one (I think) for this to allow communication and sent you an invite. Thanks for the suggestions, I'm excited to see more improvement added to the game.
